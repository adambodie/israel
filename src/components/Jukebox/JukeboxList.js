import React, { Component } from 'react';
import JukeboxItem from './JukeboxItem';

export default class JukeboxList extends Component {

    constructor(props) {
        super(props);
        this.state = {
            data: [
					{title: 'Tel Aviv', src: 'nMQw29nfzpg'},
					{title: 'Juicy', src: '_JZom_gVfuw'},	
					{title: 'Take Me Home Country Roads', src: '1vrEljMfXYo'},	
					{title: 'All Star', src: 'L_jWHffIx5E'},
					{title: 'American Idiot', src: 'Ee_uujKuJMI'},
					{title: 'Tiny Dancer', src: 'yYcyacLRPNs'},
					{title: 'Don\'t Stop Believin\'', src: '1k8craCGpgs'}
				]
			};
		}
  
    render() {
        let cards = [];
        this.state.data.forEach((y, index) => cards.push(<JukeboxItem data={y} key={index}/>));
        return (
            <div className="jukebox-flex">
                {cards}
            </div>
        );
    }

};

