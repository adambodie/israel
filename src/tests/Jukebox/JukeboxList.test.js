import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import JukeboxList from '../../components/Jukebox/JukeboxList'

it('renders without crashing', () => {
	const div = document.createElement('div')
	ReactDOM.render(<JukeboxList data={{title: 'title', src: 'src'}}/>, div)
	ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
	const tree = renderer.create(<JukeboxList data={{title: 'title', src: 'src'}}/>).toJSON()
	expect(tree).toMatchSnapshot()
})

