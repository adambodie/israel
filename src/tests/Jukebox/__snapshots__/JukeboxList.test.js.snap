// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jukebox-flex"
>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Tel Aviv
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/nMQw29nfzpg"
          title="Tel Aviv"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Juicy
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/_JZom_gVfuw"
          title="Juicy"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Take Me Home Country Roads
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/1vrEljMfXYo"
          title="Take Me Home Country Roads"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Tiny Dancer
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/yYcyacLRPNs"
          title="Tiny Dancer"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Bohemian Rhapsody
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/fJ9rUzIMcZQ"
          title="Bohemian Rhapsody"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          American Idiot
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/Ee_uujKuJMI"
          title="American Idiot"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          All Star
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/L_jWHffIx5E"
          title="All Star"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Stacy's Mom
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/dZLfasMPOU4"
          title="Stacy's Mom"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          I Want It That Way
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/4fndeDfaWCg"
          title="I Want It That Way"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Don't Stop Believin'
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/1k8craCGpgs"
          title="Don't Stop Believin'"
        />
      </div>
    </div>
  </div>
  <div
    className="flipper"
    data-testid="front-view"
  >
    <div
      className="card"
    >
      <button
        className="jukebox-button"
        onClick={[Function]}
      >
        o
      </button>
      <div
        className="front"
      >
        <p>
          Understood
        </p>
      </div>
      <div
        className="back"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/4dV4QC9qzBs"
          title="Understood"
        />
      </div>
    </div>
  </div>
</div>
`;
